packages/supermassive/src/codegen/utilities.ts

```diff
@@ -0,0 +1,240 @@
+import ts, { factory } from "typescript";
+import { ASTNode, Kind } from "graphql";
+
+export function createNullableType(node: ts.TypeNode): ts.UnionTypeNode {
+  return factory.createUnionTypeNode([
+    node,
+    factory.createLiteralTypeNode(factory.createNull()),
+  ]);
+}
+
+export function createNonNullableType(node: ts.TypeNode): ts.TypeNode {
+  if (ts.isUnionTypeNode(node)) {
+    return node.types[0];
+  } else {
+    throw new Error(`Can't make type non nullable: ${node}.`);
+  }
+}
+
+export function isDirectAncestorInput(
+  ancestors: readonly (ASTNode | readonly ASTNode[])[],
+): boolean {
+  const directAncestors = ancestors[ancestors.length - 1];
+  if (Array.isArray(directAncestors)) {
+    return directAncestors.some((directAncestor) => {
+      return (
+        "kind" in directAncestor &&
+        directAncestor.kind === Kind.INPUT_VALUE_DEFINITION
+      );
+    });
+  }
+
+  return (
+    "kind" in directAncestors &&
+    directAncestors.kind === Kind.INPUT_VALUE_DEFINITION
+  );
+}
+
+export function createVariableNameFromImport(path: string): string {
+  return camelCase(
+    path
+      .replace(/\.\.\//g, "up-")
+      .replace(/@/g, "NS-")
+      .replace(/\.\//g, "cwd-")
+      .replace(/\//g, "-"),
+    {
+      pascalCase: true,
+      preserveConsecutiveUppercase: true,
+    },
+  );
+}
+
+/**
+ * Vendored due to jest being shit and unable to cleanly use ESM modules.
```

Comment:
[censored] 😁

```suggestion
 * Vendored due to jest being unable to cleanly use ESM modules.
```

Reply:
Maybe we can use `lodash.startcase` (and additionally remove the space). It seems to be doing the same thing? https://lodash.com/docs#startCase